cfgov/v1/tests/test_util.py

```diff
@@ -19,57 +23,61 @@ def setUp(self):
-            mock.Mock(**{'content_json': {'live': False, 'shared': False}}),
-            mock.Mock(**{'content_json': {'live': False, 'shared': False}}),
-            mock.Mock(**{'content_json': {'live': False, 'shared': False}}),
-        ]
-        mock_json_loads.side_effect = [page.content_json for page in self.page_versions]
-        util.get_appropriate_page_version(self.request, self.page)
-        assert not self.page_versions[0].as_page_object.called
-        assert not self.page_versions[1].as_page_object.called
-        assert not self.page_versions[2].as_page_object.called
-        assert not self.page_versions[3].as_page_object.called
+        self.new_page = CFGOVPage(title='a cfgov page')
+        self.new_page.live = False
+        self.new_page.shared = False
 
+        helpers.save_new_page(self.new_page)
+        content_json = json.loads(self.new_page.to_json())
+        # create the various page revisions
+        # revision 1
+        content_json['title'] = 'revision 1'
+        content_json['live'] = True
+        content_json['shared'] = True
+        self.revision_1 = self.new_page.revisions.create(content_json=json.dumps(content_json))
 
-    @mock.patch('json.loads')
-    def test_shared_returns_None_if_page_not_live_when_on_production(self, mock_json_loads):
-        self.request.site.hostname = 'localhost'
-        self.page_versions = [
-            mock.Mock(**{'content_json': {'live': False, 'shared': True}}),
-            mock.Mock(**{'content_json': {'live': False, 'shared': True}}),
-            mock.Mock(**{'content_json': {'live': False, 'shared': True}}),
-            mock.Mock(**{'content_json': {'live': False, 'shared': True}}),
-        ]
-        mock_json_loads.side_effect = [page.content_json for page in self.page_versions]
-        util.get_appropriate_page_version(self.request, self.page)
-        assert not self.page_versions[0].as_page_object.called
-        assert not self.page_versions[1].as_page_object.called
-        assert not self.page_versions[2].as_page_object.called
-        assert not self.page_versions[3].as_page_object.called
+        # rev 2
+        content_json['title'] = 'revision 2'
+        content_json['live'] = True
+        content_json['shared'] = False
+        self.revision_2 = self.new_page.revisions.create(content_json=json.dumps(content_json))
+
+        # rev 3
+        content_json['title'] = 'revision 3'
+        content_json['live'] = False
+        content_json['shared'] = True
+        self.revision_3 = self.new_page.revisions.create(content_json=json.dumps(content_json))
+
+        # rev 4
+        content_json['title'] = 'revision 4'
+        content_json['live'] = False
+        content_json['shared'] = False
+        self.revision_4 = self.new_page.revisions.create(content_json=json.dumps(content_json))
+
+    def test_production_returns_first_live_page(self):
+        self.request.is_staging = False
```

Comment:
I'm not really sure why this is hard to understand. These objects are attached to `self` because they are carried over from test to test; they are initialized in `setUp` so it's pretty clear what they're for. I prefer the method of factoring out common operations into `setUp` because it lets you see what is the same across all tests. For this particular series of tests, I don't feel super strongly about it and I can change it if people really prefer @chosak's version, but honestly I don't think it really improves readability in any meaningful way.